crates/bevy_pbr/src/prepass/mod.rs

```diff
@@ -568,43 +792,97 @@ pub fn prepare_prepass_textures(
+                            usage: TextureUsages::RENDER_ATTACHMENT
+                                | TextureUsages::TEXTURE_BINDING,
+                            view_formats: &[],
+                        },
+                    )
+                })
+                .clone()
+        });
+
         commands.entity(entity).insert(ViewPrepassTextures {
             depth: cached_depth_texture,
             normal: cached_normals_texture,
+            motion_vectors: cached_motion_vectors_texture,
             size,
         });
     }
 }
 
 #[derive(Default, Resource)]
 pub struct PrepassViewBindGroup {
-    bind_group: Option<BindGroup>,
+    motion_vectors: Option<BindGroup>,
+    no_motion_vectors: Option<BindGroup>,
 }
 
 pub fn queue_prepass_view_bind_group<M: Material>(
     render_device: Res<RenderDevice>,
     prepass_pipeline: Res<PrepassPipeline<M>>,
     view_uniforms: Res<ViewUniforms>,
     globals_buffer: Res<GlobalsBuffer>,
+    previous_view_proj_uniforms: Res<PreviousViewProjectionUniforms>,
     mut prepass_view_bind_group: ResMut<PrepassViewBindGroup>,
 ) {
-    let Some(view_binding) = view_uniforms.uniforms.binding() else { return };
-    let Some(globals_binding) = globals_buffer.buffer.binding() else { return };
-    prepass_view_bind_group.bind_group =
-        Some(render_device.create_bind_group(&BindGroupDescriptor {
-            entries: &[
-                BindGroupEntry {
-                    binding: 0,
-                    resource: view_binding,
-                },
-                BindGroupEntry {
-                    binding: 1,
-                    resource: globals_binding,
-                },
-            ],
-            label: Some("prepass_view_bind_group"),
-            layout: &prepass_pipeline.view_layout,
-        }));
+    if let (Some(view_binding), Some(globals_binding), Some(previous_view_proj_binding)) = (
+        view_uniforms.uniforms.binding(),
+        globals_buffer.buffer.binding(),
+        previous_view_proj_uniforms.uniforms.binding(),
+    ) {
+        prepass_view_bind_group.motion_vectors =
+            Some(render_device.create_bind_group(&BindGroupDescriptor {
+                entries: &[
+                    BindGroupEntry {
+                        binding: 0,
+                        resource: view_binding,
+                    },
+                    BindGroupEntry {
+                        binding: 1,
+                        resource: globals_binding,
+                    },
+                    BindGroupEntry {
+                        binding: 2,
+                        resource: previous_view_proj_binding,
+                    },
+                ],
+                label: Some("prepass_view_motion_vectors_bind_group"),
+                layout: &prepass_pipeline.view_layout_motion_vectors,
+            }));
+    }
+
+    if let (Some(view_binding), Some(globals_binding)) = (
+        view_uniforms.uniforms.binding(),
+        globals_buffer.buffer.binding(),
+    ) {
```

Comment:
Could do this if let once, and then within it, do an if let for the preview view proj binding.